nodejs/image.bzl

```diff
@@ -69,34 +79,23 @@ def _dep_layer_impl(ctx):
     return dep_layer_impl(ctx, runfiles = _runfiles, emptyfiles = _emptyfiles)
 
 _dep_layer = rule(
-    attrs = dict(_container.image.attrs.items() + {
-        # The base image on which to overlay the dependency layers.
-        "base": attr.label(mandatory = True),
-        # The dependency whose runfiles we're appending.
-        "dep": attr.label(
-            mandatory = True,
-            allow_files = True,
-        ),
-
-        # Whether to lay out each dependency in a manner that is agnostic
-        # of the binary in which it is participating.  This can increase
-        # sharing of the dependency's layer across images, but requires a
-        # symlink forest in the app layers.
-        "agnostic_dep_layout": attr.bool(default = True),
-        # The binary target for which we are synthesizing an image.
-        # This is needed iff agnostic_dep_layout.
-        "binary": attr.label(mandatory = False),
-
-        # Override the defaults.
-        # https://github.com/bazelbuild/bazel/issues/2176
-        "data_path": attr.string(default = "."),
-        "directory": attr.string(default = "/app"),
-    }.items()),
+    attrs = dep_layer_attrs,
     executable = True,
     outputs = _container.image.outputs,
     implementation = _dep_layer_impl,
 )
 
+def _app_layer_impl(ctx):
+    empty_dirs = ["/".join(["/app", ctx.workspace_name])]
```

Comment:
can you document here briefly why the empty dirs need to be added (and anything else specific about this impl that makes it different from the one in lang/image.bzl, e.g., what you are doing with the runfiles)